Integrations/ARIA/aria_packet_intelligence.py

```diff
@@ -0,0 +1,1645 @@
+            trid: The request id when you want to adding a rule to ARIA PI Reaper.
+
+        Returns: True if complete, False if not.
+
+        """
+
+        # url to valid the request
+        trid_url = self.sdso_url + f'/packetClassification/completion/transaction?PC_TRID={trid}'
+
+        t0 = time.perf_counter()
+
+        delta = time.perf_counter() - t0
+
+        while delta < 20:
+            res = requests.get(trid_url, timeout=self.time_out)
+
+            delta = time.perf_counter() - t0
+
+            if res:
+                info = res.json()
+                for tcl_entry in info['tclList']:
+                    if 'SUCCESS' in tcl_entry['status']:
+                        return True
+                    elif 'FAILURE' in tcl_entry['status']:
+                        return False
+
+            time.sleep(0.1)
+
+        return False
+
+    def _remove_rule(self, rule_name: str, logic_block: str, instance_id: str = None, label_sia_group: str = None,
+                     label_sia_name: str = None, label_sia_region: str = None) -> dict:
+        """ Remove rule in the ARIA PI Reaper
+
+        Args:
+            rule_name: The name of the rule to create.
+            logic_block: Parameter used to form named rule data. Examples: '5-tuple', 'src-port', etc.
+            instance_id: The instance number of the ARIA PI instance.
+            label_sia_group: The name of the group to which the SIA belongs.
+            label_sia_name: The name of the SIA.
+            label_sia_region: The name of the region to which SIA belongs.
+
+        Returns: Dictionary context data contains useful response information.
+
+        """
+
+        url = self.sdso_url + '/ruleForward'
+
+        headers = {'Content-type': 'application/json', 'Accept': 'text/plain'}
+
+        data = self._generate_named_rule_data(rule_name, logic_block, '', 'remove', instance_id,
+                                              label_sia_group, label_sia_name, label_sia_region)
+
+        response = requests.put(url, data=json.dumps(data), headers=headers, timeout=self.time_out)
+
+        response_timestamp = ''
+
+        endpoints = []
+
+        command_state = False
+
+        if response.ok:
+            error_info = ''
+
+            command_state = True
+
+            endpoints = response.json()['endpoints']
+
+            for ep in endpoints:
+                trid = ep['trid']
+                success = self._wait_for_trid(trid)
+                ep['completion'] = success
+                if not success:
+                    command_state = False
+            response_timestamp = response.json()['timestamp']
+        else:
+            error_info = response.text
+
+        if command_state:
+            command_state_str = 'Success'
+        else:
+            command_state_str = 'Failure'
+
+        context_data = {
+            'Rule': {
+                'Name': rule_name,
+                'Definition': ''
+            },
+            'Status': {
```

Comment:
Don't think we need this in the outputs, ok with you to remove it ?

Reply:
Which one should be removed? Status or definition?

Reply:
Status

Reply:
Can I still keep it? Since I feel it is useful.

Reply:
you can keep it if you want, even though it looks a bit weird to get the api response status in the outputs (we are not doing it in any integration), since in the outputs there should be only stuff that can be used in a playbook as input for another task

Reply:
Oh, I understand, I think I can remove error_info I think this won't be used. The request status code is actually necessary. The status is necessary in our Rest API, since if the status fails, users can changes some query variables to find a new spot. And status_code means it returns a trid id. But the "work" might not finished if there is another rules already there.

Reply:
👍